library/archive.py

```diff
@@ -0,0 +1,89 @@
+import os
+from pathlib import Path
+
+from . import (
+    aws_access_key_id,
+    aws_s3_bucket,
+    aws_s3_endpoint,
+    aws_secret_access_key,
+    base_path,
+    pp,
+)
+from .ingest import Ingestor
+from .s3 import S3
+
+
+class Archive:
+    def __init__(self):
+        """
+        Initialize the Archive class instance
+        """
+        self.ingestor = Ingestor()
+        self.s3 = S3(
+            aws_access_key_id, aws_secret_access_key, aws_s3_endpoint, aws_s3_bucket
+        )
+
+    def __call__(
+        self,
+        path: str,
+        output_format: str,
+        push: bool = False,
+        clean: bool = False,
+        latest: bool = False,
+        name: str = None,
```

Comment:
Add a description of this parameter in the docstring. It might also be worth using the name technique (rather than path) for one of the examples in sample usage.